semgrep/semgrep/output.py

```diff
@@ -429,7 +429,12 @@ def close(self) -> None:
                 self.settings.output_per_finding_max_lines_limit,
             )
             if output:
-                print(output, file=self.stdout)
+                try:
+                    print(output, file=self.stdout)
+                except UnicodeEncodeError as ex:
+                    raise Exception(
```

Comment:
I would say this isn't the more ergonomic solution, but it's the best I could do without major changes to our error and output functionality.